microsoft/testsuites/kata_conformance/kata_conformance_tests.py

```diff
@@ -0,0 +1,117 @@
+        description="""
+            Runs Kubernetes Kata Cotaniner Conformance test.
+        """,
+        priority=3,
+        timeout=36000,
+    )
+    def kata_conformance_tests(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        # ensure virtualization is enabled in hardware before running tests
+        virtualization_enabled = node.tools[Lscpu].is_virtualization_enabled()
+        if not virtualization_enabled:
+            raise SkippedException("Virtualization is not enabled in hardware")
+        if not isinstance(node.os, (CBLMariner, Ubuntu)):
+            raise SkippedException(
+                f"Kata Conformance tests are not implemented in LISA for {node.os.name}"
+            )
+
+        log.info("Creating AKS Infra")
+        log.debug("Variables " + json.dumps(variables))
+        subscription_id = validate_variable(
+            variables, "subscription_id", log, nullable=True
+        )
+        client_id = validate_variable(
+            variables, "service_principal_client_id", log, nullable=True
+        )
+        client_secret = validate_variable(
+            variables, "service_principal_key", log, nullable=True
+        )
+        tenant_id = validate_variable(
+            variables, "service_principal_tenant_id", log, nullable=True
+        )
+        kubernetes_version = validate_variable(variables, "kubernetes_version", log)
+        worker_vm_size = validate_variable(variables, "worker_vm_size", log)
+        node_count = validate_variable(variables, "node_count", log)
+        azure_region = validate_variable(variables, "azure_region", log)
+
+        headers = {
+            "AKSHTTPCustomFeatures": validate_variable(
+                variables, "aks_http_custom_features", log
+            ),
+            "OSImageSubscriptionID": validate_variable(
+                variables, "os_image_subscription_id", log
+            ),
+            "OSImageResourceGroup": validate_variable(
+                variables, "os_image_resource_group", log
+            ),
+            "OSImageGallery": validate_variable(variables, "os_image_gallery", log),
+            "OSImageName": validate_variable(variables, "os_image_name", log),
+            "OSImageVersion": validate_variable(variables, "os_image_version", log),
+            "OSSKU": validate_variable(variables, "ossku", log),
+        }
+
+        aks = AKSInfra(
+            log=log,
+            subscription_id=subscription_id,
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant_id=tenant_id,
+        )
+        aks.create_aks_infra(
+            kubernetes_version, worker_vm_size, node_count, azure_region, headers
+        )
+        log.info("Completed AKS Infra Creation")
+
+        node.tools[KataConformanceTests].run_tests(result, environment, log_path)
```

Comment:
How is the node related to the aks cluster?

Reply:
This Node will launch the kata conformance test case on AKS cluster. While creating AKS cluster, we are creating Kubernetes config file at ~/.kube/config. 

After building Kubernetes, we will use binary e2e.test to launch conformance test on AKS using above Kubernetes config file.